server/py/services/api/initial_data.py

```diff
@@ -877,40 +877,84 @@ def _perform_version_8_data_migrations(
 def _perform_version_9_data_migrations(
     db: framework.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
 ):
-    _migrate_function_kind(db, db_session, chunk_size=500)
+    _migrate_data(
+        db,
+        db_session,
+        framework.db.sqldb.models.Function,
+        "kind",
+        _handle_function_kind,
+    )
+    _migrate_data(
+        db,
+        db_session,
+        framework.db.sqldb.models.ArtifactV2,
+        "producer_uri",
+        _handle_artifact_producer_uri,
+    )
+
 
+def _handle_function_kind(records: list, record):
+    function_dict = record.struct
+    record.kind = function_dict.pop("kind", None)
```

Comment:
I know this isn't part of this PR, but @liranbg - why are you `pop`-ing the field from the dictionary? You'll need to re-introduce it into the dictionary when retrieving objects. Any reason for that?

Reply:
to avoid data duplication. I re-add it back when you retrieve the function

Reply:
it is to avoid data duplication, we discussed it on his PR and he added unit tests to ensure it is reintroduced on pulls

Reply:
I'm not sure why data duplication is an issue here. Most of the fields in the DB are also part of the dictionary to begin with (project, name etc.). Making this a one-time behavior doesn't make the code any clearer, as it's extra logic that is unique to this case.

Reply:
you're stating a problem, not a solution. since I do not intend to preserve the correctness of both body/column data, ive set the value to be taken from the column and enrich the body at GET time to ensure smooth BC.